tableauserverclient/models/permissions_item.py

```diff
@@ -0,0 +1,151 @@
+import xml.etree.ElementTree as ET
+
+
+class ItemWithPermissions(object):
+    pass
+
+
+class Permission:
+    class Type:
+        Datasource = 'datasource'
+        Workbook = 'workbook'
+        Project = 'project'
+
+    class GranteeType:
+        User = 'user'
+        Group = 'group'
+
+    class CapabilityMode:
+        Allow = 'Allow'
+        Deny = 'Deny'
+
+    class DatasourceCapabilityType:
+        ChangePermissions = 'ChangePermissions'
+        Connect = 'Connect'
+        Delete = 'Delete'
+        ExportXml = 'ExportXml'
+        Read = 'Read'
+        Write = 'Write'
+
+    class WorkbookCapabilityType:
+        AddComment = 'AddComment'
+        ChangeHierarchy = 'ChangeHierarchy'
+        ChangePermissions = 'ChangePermissions'
+        Delete = 'Delete'
+        ExportData = 'ExportData'
+        ExportImage = 'ExportImage'
+        ExportXml = 'ExportXml'
+        Filter = 'Filter'
+        Read = 'Read'
+        ShareView = 'ShareView'
+        ViewComments = 'ViewComments'
+        ViewUnderlyingData = 'ViewUnderlyingData'
+        WebAuthoring = 'WebAuthoring'
+        Write = 'Write'
+
+    class ProjectCapabilityType:
+        ProjectLeader = 'ProjectLeader'
+        Read = 'Read'
+        Write = 'Write'
+
+
+class GranteeCapabilityItem(object):
+    def __init__(self, grantee_type=None, grantee_id=None, capabilities=None):
+        self._grantee_type = grantee_type
+        self._grantee_id = grantee_id
+        self._capabilities = capabilities
+
+    def _set_values(self, grantee_type, grantee_id, capabilities):
+        self._grantee_type = grantee_type
+        self._grantee_id = grantee_id
+        self._capabilities = capabilities
+
+    @property
+    def grantee_type(self):
+        return self._grantee_type
+
+    @property
+    def grantee_id(self):
+        return self._grantee_id
+
+    @property
+    def capabilities(self):
+        return self._capabilities
+
+
+class PermissionsItem(object):
+    def __init__(self):
+        self._type = None
+        self._item_id = None
+        self._grantee_capabilities = None
+
+    def _set_values(self, type, item_id, grantee_capabilities):
+        self._type = type
+        self._item_id = item_id
+        self._grantee_capabilities = grantee_capabilities
+
+    @property
+    def type(self):
+        return self._type
+
+    @property
+    def item_id(self):
+        return self._item_id
+
+    @property
+    def grantee_capabilities(self):
+        return self._grantee_capabilities
+
+    @property
+    def is_user_permission(self):
+        return self._user_id is not None
+
+    @property
+    def is_group_permission(self):
+        return self._group_id is not None
+
+    @classmethod
+    def from_response(cls, resp, ns=None):
+        permissions = PermissionsItem()
+        parsed_response = ET.fromstring(resp)
+
+        for option in ('workbook', 'datasource', 'project'):
```

Comment:
Sure, it looks like it might be possible to pull the first element within the parent but that could be risky across different versions.  I can definitely pull into a constant.